File: src/main/java/com/core/base/corebase/controller/review/ReviewController.java
//// Copyright 2025 Kakao Healthcare Corp. All Rights Reserved.
//
//package com.core.base.corebase.controller.review;
//
//import com.core.base.corebase.controller.review.dto.ReviewReq;
//import com.core.base.corebase.controller.review.dto.ReviewRes;
//import com.core.base.corebase.controller.review.dto.ReviewSurveyRes;
//import com.core.base.corebase.service.review.ReviewService;
//import com.core.base.corebase.service.review.SurveyService;
//import io.swagger.v3.oas.annotations.tags.Tag;
//import lombok.RequiredArgsConstructor;
//import org.springframework.web.bind.annotation.*;
//
//import java.util.UUID;
//
//@Tag(name = "리뷰 API")
//@RestController
//@RequestMapping("/reviews")
//@RequiredArgsConstructor
//public class ReviewController {
//
//    private final ReviewService reviewService;
//    private final SurveyService surveyService;
//
//    @PostMapping
//    public void save(@RequestBody ReviewReq req) {
//        reviewService.save(req);
//    }
//
//    @GetMapping("/{id}")
//    public ReviewRes get(@PathVariable Long id) {
//        return reviewService.get(id);
//    }
//
//    @PostMapping("/{id}/pause")
//    public void pause(@PathVariable UUID id) {
//        reviewService.pause(id);
//    }
//
//    @GetMapping("/{id}/surveys")
//    public ReviewSurveyRes getSurvey(@PathVariable UUID id) {
//        return surveyService.get(id);
//    }
//}

File: src/main/kotlin/com/core/base/corebase/controller/review/ReviewController.kt
package com.core.base.corebase.controller.review

import com.core.base.corebase.controller.review.dto.ReviewReq
import com.core.base.corebase.controller.review.dto.ReviewRes
import com.core.base.corebase.controller.review.dto.ReviewSurveyRes
import com.core.base.corebase.service.review.ReviewService
import com.core.base.corebase.service.review.SurveyService
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.*
import java.util.*


@Tag(name = "리뷰 API")
@RestController
@RequestMapping("/reviews")
class ReviewController(
    private val reviewService: ReviewService,
    private val surveyService: SurveyService
) {

    @PostMapping
    fun save(@RequestBody req: ReviewReq) = reviewService.save(req)

    @GetMapping("/{id}")
    fun get(@PathVariable id: UUID) : ReviewRes = reviewService.get(id)

//    @GetMapping("/{id}/reviewee/{uid}")
//    fun getReviewByReviewee(@PathVariable(name = "id") id: UUID, @PathVariable(name = "uid") uid: UUID) : ReviewDetailRes = reviewService.getReview(id, uid)
//
//    @GetMapping("/reviewee/{uid}")
//    fun listReviewByReviewee(@PathVariable uid: UUID) : List<ReviewerRes> = reviewService.listReviewByReviewee(uid)
//
//    @GetMapping("/reviewer/{uid}")
//    fun listReviewByReviewer(@PathVariable uid: UUID) : List<ReviewerRes> = reviewService.listReviewByReviewer(uid)

    @PostMapping("/{id}/pause")
    fun pause(@PathVariable id: UUID) = reviewService.pause(id)

    @GetMapping("/{id}/surveys")
    fun getSurvey(@PathVariable id: UUID) : ReviewSurveyRes = surveyService.get(id)

//    @PostMapping("/{id}/surveys")
//    fun saveSurvey(@PathVariable id: UUID, @RequestBody req: ReviewSurveyReq) = surveyService.save(id, req)

}